### PR TITLE
Simplify name of trino-cpp modules and library names

### PR DESCRIPTION
**Pull request by jikunshang**
rename some modules and library names, align with document

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - uses: actions/checkout@v3
       - name: build cpp
         run: |
-          cd trino-cpp
+          cd Gluten-Trino-Cpp
```
Comment (winningsix) [inline .github/workflows/test.yaml:26]: Hmm, will cpp be more simple as the repo name already specifics Gluten-Trino?

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,3 @@
-[submodule "trino-cpp/velox"]
-	path = trino-cpp/velox
+[submodule "Gluten-Trino-Cpp/velox"]
```
Comment (winningsix) [inline .gitmodules:1]: Same as above.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (jikunshang) [reply .github/workflows/test.yaml:26]: maybe `cpp` as a dir name a little confusing?

Comment (winningsix) [reply .github/workflows/test.yaml:26]: How do we plan to name ```GlutenTrinoPlugin``` folder? Two options on top of mind: 1. top level folders as java, cpp like [Arrow](https://github.com/apache/arrow); 2. Gluten-like, which root folder has a maven pom. I am more preferring to use option 1. 

Thoughts?

Comment (jikunshang) [reply .github/workflows/test.yaml]: renamed as option 1

**Pull request merged**